examples/auth.py

```diff
@@ -0,0 +1,78 @@
+import os
+
+import grpc
+import google.auth.transport.requests
+import google.auth.transport.grpc
+
+from kessel.grpc.auth import OIDCDiscovery, OAuth2ClientCredentials
+from kessel.inventory.v1beta2 import (
+    check_request_pb2,
+    inventory_service_pb2_grpc,
+    reporter_reference_pb2,
+    resource_reference_pb2,
+    subject_reference_pb2,
+)
+
+KESSEL_ENDPOINT = os.environ.get("KESSEL_ENDPOINT", "localhost:9000")
+ISSUER_URL = os.environ.get("AUTH_DISCOVERY_ISSUER_URL", "")
+CLIENT_ID = os.environ.get("AUTH_CLIENT_ID", "")
+CLIENT_SECRET = os.environ.get("AUTH_CLIENT_SECRET", "")
+
+
+def run():
+    try:
+        discovery = OIDCDiscovery(ISSUER_URL)
+        token_endpoint = discovery.fetch_oidc_discovery()  # Network call occurs here
```

Comment:
I would flip this slightly –

```py
    discovery = fetchOIDCDiscovery(issuer_url)
    token_endpoint = discovery.token_endpoint
```

The difference is:

- the discovery object is just the current state of metadata, not a service to fetch metadata
- the discovery is not just the token endpoint; it has a lot of useful metadata. We don't need to model all of that–we can literally just have the token_endpoint parsed and that's it. But, what it looks like here is that "discovery" is _the same as_ the token endpoint which is not true (OIDC discovery is a well specified thing)

Does that make sense?

Reply:
Ahhh yeah okay I was basically just thinking of it as a service to get the endpoint explicitly, and just throwing everything else out. 

I see so like a top level function for actually fetching the discovery document which can have a property of the `token_endpoint`?